src/datatypes/service-context.ts

```diff
@@ -16,20 +16,21 @@
  */
 
 import { AccountConfig } from './account-config';
+import ServiceConfig from './service-config';
 
-export class ServiceContext {
+export class ServiceContext<Config extends ServiceConfig = any> {
```

Comment:
This and the below changes let us optionally specify what type the params will be, but it leaves all current usages intact and unmodified. I really, really like having default values for types!